model/task/task.go

```diff
@@ -658,9 +661,68 @@ func (t *Task) MarkAsDispatched(hostId, distroId, agentRevision string, dispatch
 			return t.DisplayTask.MarkAsDispatched("", "", "", dispatchTime)
 		}
 	}
+
 	return nil
 }
 
+func (t *Task) CalculateUnblockedTime() error {
```

Comment:
Because this function makes an actual change and doesn't return a calculated variable, i'd suggest calling it something like "UpdateUnblockedTime"